### PR TITLE
Add hooks for adding interceptors.

### PR DESCRIPTION
**Pull request by sfc-gh-jchacon**
Either when using the server build helper or the cmd libraries.

This way users can append their own interceptors for telemetry/etc without having to worry we're forcing a certain usage.

The only thing we force is logging and OPA. Though likely logging should be made more customizable too.

### CODE REVIEW

File: cmd/proxy-server/server/server.go [hunk 1]
```diff
@@ -55,6 +55,18 @@ type RunState struct {
 	// entry is found. The supplied function can then do any validation it wants
 	// in order to ensure it's compliant.
 	JustificationFunc func(string) error
+	// UnaryInterceptors are any additional interceptors to be added to unary RPCs
```
Comment (sfc-gh-jallie) [inline cmd/proxy-server/server/server.go:58]: optional: how do you feel about the OptionFunc pattern? 

I've grown to appreciate it vs "option structs", but I realize it's a matter of taste. It does have the advantage of mirroring the option style that gRPC uses. 

we could probably refactor this into something like

```go
type runState {
  // existing runstate fields
}

type Option interface {
   apply(*runState) error
} 

type optionFunc func(*runstate) error

func (o optionFunc) apply(r *runstate) error {
  return o(r)
}

func WithLogger(l logr.Logger) Option {
  return optionFunc(func(r *runState) error {
      r.Logger = l
      return nil
  })
}

// ...other options

func (s *Server) Run(ctx context.Context, opts ...Option) {
  rs := &runState{}
  for _, o := range opts {
     o.apply(rs)
  }
 // use runstate here
}
```

**Review by sfc-gh-jallie — APPROVED** [review]
LGTM

File: cmd/proxy-server/server/server.go (hunk 1 above)
Comment (sfc-gh-jchacon) [reply cmd/proxy-server/server/server.go:58]: I like this and agree we should go this route. Let me rework

Comment (sfc-gh-jchacon) [reply cmd/proxy-server/server/server.go]: Take another look please for sanity.

**Review by sfc-gh-jallie — APPROVED** [review]
I think this looks much nicer at the call sites

**Pull request merged**